web/src/features/playground/client/context/index.tsx

```diff
@@ -73,10 +73,10 @@ export const PlaygroundProvider: React.FC<PropsWithChildren> = ({
 
   const { data: initialPrompt, isInitialLoading } = api.prompts.byId.useQuery(
     {
-      projectId,
+      projectId: projectId as string, // Typecast as query is enabled only when projectId is present
       id: initialPromptId ?? "",
     },
-    { enabled: Boolean(initialPromptId), staleTime: Infinity }, // do not refetch as this would overwrite the user's input
+    { enabled: Boolean(initialPromptId && projectId), staleTime: Infinity }, // do not refetch as this would overwrite the user's input
```

Comment:
```suggestion
    { enabled: !!initialPromptId && !!projectId, staleTime: Infinity }, // do not refetch as this would overwrite the user's input
```

I think Boolean(..) can be confusing here and unsure how truthines works

Reply:
no strong opinion here, I was unsure about `initialPromptId &&` in case of e.g. `""`. Just confirmed that this does not matter at all. resolving this